client/rest/src/test/java/org/elasticsearch/client/RestClientSingleHostIntegTests.java

```diff
@@ -76,8 +74,7 @@ public static void startHttpServer() throws Exception {
         }
 
         httpServer = createHttpServer();
-        int numHeaders = randomIntBetween(0, 5);
-        defaultHeaders = generateHeaders("Header-default", "Header-array", numHeaders);
+        defaultHeaders = RestClientTestUtil.randomHeaders(getRandom(), "Header-default");
```

Comment:
Also mentioned below, I don't know which purpose array name ("Header-array") served here before, just curious to see why its gone (unnecessary?).